runner/runner/listener.py

```diff
@@ -59,6 +62,107 @@ def _handle_delivery(channel, deliver, properties, body):
             case _:
                 logger.error("Unrecognized message type: %s", msg_type)
 
-        channel.basic_ack(deliver.delivery_tag)
+        channel.basic_ack(method.delivery_tag)
     except Exception as exc:
         logger.error("Error handling received message: %s", exc)
+
+
+def _get_inspect() -> Inspect:
+    """Return Celery Inspect object after connection to Celery Workers is established"""
+    inspect = Inspect(app=app)
+    while (workers := inspect.ping()) is None:
+        logger.info("Waiting for connection to Celery workers to be established.")
+        sleep(1)
+
+    logger.info(f"Number of available workers: {len(workers.keys())}")
+    return inspect
+
+
+def _create_inspect() -> None:
+    """Create global Celery Inspect object to interact with Celery Worker API"""
+    global celery_inspect
```

Comment:
I don't think there's a compelling reason to create a global here.  Other approaches you might take:

* Using `@cached` on `_get_inspect()` so that it only does the wait for connection once.
* Storing the result of `_get_inspect()` and just passing it into the `_wait_for_available_workers()` call:

```python
inspect = _get_inspect()

while True:
    method, properties, body = channel.basic_get("public")

    if method is None:
        sleep(WAIT_FOR_MESSAGE_DELAY)
        continue

    _handle_delivery(channel, method, properties, body)
    _wait_for_available_worker(inspect)
```